kartothek/io/dask/delayed.py

```diff
@@ -129,55 +127,46 @@ def garbage_collect_dataset__delayed(
     )
 
 
-def _load_and_merge_mps(mp_list, store, label_merger, metadata_merger, merge_tasks):
-    mp_list = [mp.load_dataframes(store=store) for mp in mp_list]
+def _load_and_merge_mps(
+    mp_list, store, label_merger, metadata_merger, merge_tasks, predicates, columns
+):
+    mp_list = [
+        mp.load_dataframes(store=store, columns=columns[mp.dataset_metadata.uuid])
+        for mp in mp_list
+    ]
     mp = MetaPartition.merge_metapartitions(
         mp_list, label_merger=label_merger, metadata_merger=metadata_merger
     )
     mp = mp.concat_dataframes()
 
     for task in merge_tasks:
-        mp = mp.merge_dataframes(**task)
+        mp = mp.merge_dataframes(**task, predicates=predicates)
 
-    return mp
+    return mp.data["table"]
 
 
 @default_docs
 def merge_datasets_as_delayed(
-    left_dataset_uuid,
-    right_dataset_uuid,
```

Comment:
This is an API-breaking change, either add a new function or add a new attribute but don't modify existing ones.